doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+* Get PFC to DSCP map
+	```combine dscp_to_tc with dscp_to_tc_keys```
+
+
+### pfc_storm_template (scope="module")
+Creates dictionary which items will be used to start/stop PFC generator on fanout switch.
+Dictionary values depends on fanout HWSKU (MLNX-OS, Arista or others)
+
+*Return dictionary in format*
+```
+{"file": {
+    "pfc_storm_start": [VALUE],
+    "pfc_storm_stop": [VALUE]
+  },
+ "template_params": {
+      "pfc_gen_file": PFC_GEN_FILE,
+      "pfc_queue_index": PFC_QUEUE_INDEX,
+      "pfc_frames_number": PFC_FRAMES_NUMBER,
+      "pfc_fanout_interface": [VALUE],
+      "ansible_eth0_ipv4_addr": [VALUE],
+      "pfc_asym": True
+  }
+}
+```
+
+For HWSKU equal to "MLNX-OS":
+
+      pfc_storm_start = "pfc_storm_mlnx.j2"
+      pfc_storm_stop = "pfc_storm_stop_mlnx.j2"
+
+For HWSKU equal to "Arista":
+
+      pfc_storm_start = "pfc_storm_arista.j2"
+      pfc_storm_stop = "pfc_storm_stop_arista.j2"
+
+
+### pfc_storm_runner (scope="function")
+
+*Setup steps*
+
+* Start PFC generator on fanout switch
+```
+fanout.exec_template(pfc_storm_template["pfc_storm_start"], pfc_storm_template["template_params"])
+```
+
+*Teardown steps*
+
+* Stop PFC generator on fanout switch
+```
+fanout.exec_template(pfc_storm_template["pfc_storm_stop"], pfc_storm_template["template_params"])
+```
+
+ 
+### enable_pfc_asym (scope="function")
+
+*Setup steps*
+
+* Enable asymmetric PFC on all server interfaces
+
+For every server port(setup["server_ports"]) execute command:
+
+```config interface pfc asymmetric [SERVER_PORT] on```
+
+*Teardown steps*
+
+* Disable asymmetric PFC on all server interfaces
+
+For every server port(setup["server_ports"]) execute command:
+
+```config interface pfc asymmetric [SERVER_PORT] off```
+
+## Test
+
+## Test cases
+
+### Test case # 1 – Asymmetric PFC Off Generate PFC frames
+#### Test objective
+Verify that DUT generates PFC frames only on lossless priorities when asymmetric PFC is disabled
+
+#### Used fixtures
+setup
+
+#### Test steps
+- Setup:
+  - Start ARP responder
+  - Limit maximum bandwith rate on the destination port by setting "1" into SAI port attribute SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
+  
+- Clear all counters for all ports
+- Get lossless priorities
+- Get lossy priorities
+- Get server ports info
+- Get non server port info (Portchannel peers)
+- Send 10000 packets for lossless priorities from all server ports (src) to non-server port (dst)
+- Verify that some packets are dropped on src ports, which means that Rx queue is full
+- Verify that PFC frames are generated for lossless priorities from src ports
+- Send 10000 packets for lossy priorities from all server ports (src) to non-server port (dst)
+- Verify that PFC frames are not generated for lossy priorities
+
+- Teardown:
+  - Restore maximum bandwith rate on the destination port by setting "0" into SAI port attribute SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
+   - Stop ARP responder
+
+### Test case # 2 – Asymmetric PFC Off RX pause frames
```

Comment:
Please be more consistent with test cases titles and use same naming format

Reply:
Fixed